In [1]:
import pandas as pd

In [2]:
titanic_df = pd.read_csv("Titanic_data.csv")

In [7]:
#dataframe.value_counts()
titanic_df.value_counts('Survived',normalize = True,ascending = True)

Survived
1    0.383838
0    0.616162
Name: proportion, dtype: float64

In [10]:
titanic_df[titanic_df['Sex']=="female"].value_counts('Survived') 

Survived
1    233
0     81
Name: count, dtype: int64

In [11]:
titanic_df[['Survived','Sex']].value_counts(sort = True)

Survived  Sex   
0         male      468
1         female    233
          male      109
0         female     81
Name: count, dtype: int64

In [13]:
titanic_df.value_counts(['Sex','Survived'],normalize = True)

Sex     Survived
male    0           0.525253
female  1           0.261504
male    1           0.122334
female  0           0.090909
Name: proportion, dtype: float64

Interestingly survival rate of femaleis near to death rate of male.

In [19]:
# dataframe_nunique() and dataframe.unique() returns unique elements.
titanic_df['Embarked'].unique()


array(['S', 'C', 'Q', nan], dtype=object)

In [17]:
titanic_df['Embarked'].nunique() #no of unique elements

3

In [23]:
titanic_df.isna() #checks the null values and returns true for null values

PassengerId  Survived  Pclass   Name    Sex    Age  SibSp  Parch  Ticket  \
0          False     False   False  False  False  False  False  False   False   
1          False     False   False  False  False  False  False  False   False   
2          False     False   False  False  False  False  False  False   False   
3          False     False   False  False  False  False  False  False   False   
4          False     False   False  False  False  False  False  False   False   
..           ...       ...     ...    ...    ...    ...    ...    ...     ...   
886        False     False   False  False  False  False  False  False   False   
887        False     False   False  False  False  False  False  False   False   
888        False     False   False  False  False   True  False  False   False   
889        False     False   False  False  False  False  False  False   False   
890        False     False   False  False  False  False  False  False   False   

      Fare  Cabin  Embarked  
0    False   True     False  
1    False  False     False  
2    False   True     False  
3    False  False     False  
4    False   True     False  
..     ...    ...       ...  
886  False   True     False  
887  False  False     False  
888  False   True     False  
889  False  False     False  
890  False   True     False  

[891 rows x 12 columns]

In [24]:
# dataframe.isna().sum() returns the number of null values in each column.
titanic_df.isna().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

In [26]:
# dataframe.group(by=..)
group_by_gender = titanic_df[['Sex','Fare','Age','Survived']].groupby(by='Sex')

In [27]:
group_by_gender.mean()

Fare        Age  Survived
Sex                                   
female  44.479818  27.915709  0.742038
male    25.523893  30.726645  0.188908

In [28]:
list(group_by_gender)

[('female',
          Sex     Fare   Age  Survived
  1    female  71.2833  38.0         1
  2    female   7.9250  26.0         1
  3    female  53.1000  35.0         1
  8    female  11.1333  27.0         1
  9    female  30.0708  14.0         1
  ..      ...      ...   ...       ...
  880  female  26.0000  25.0         1
  882  female  10.5167  22.0         0
  885  female  29.1250  39.0         0
  887  female  30.0000  19.0         1
  888  female  23.4500   NaN         0
  
  [314 rows x 4 columns]),
 ('male',
        Sex     Fare   Age  Survived
  0    male   7.2500  22.0         0
  4    male   8.0500  35.0         0
  5    male   8.4583   NaN         0
  6    male  51.8625  54.0         0
  7    male  21.0750   2.0         0
  ..    ...      ...   ...       ...
  883  male  10.5000  28.0         0
  884  male   7.0500  25.0         0
  886  male  13.0000  27.0         0
  889  male  30.0000  26.0         1
  890  male   7.7500  32.0         0
  
  [577 rows x 4 columns])]

In [45]:
group_by_embarked = titanic_df[['Fare','Age','Survived','Embarked']].groupby(by='Embarked')

In [46]:
group_by_embarked.sum()

Fare       Age  Survived
Embarked                                
C         10072.2962   4005.92        93
Q          1022.2543    786.50        30
S         17439.3988  16312.75       217

In [47]:
group_by_embarked = titanic_df[['Sex','Fare','Age','Survived','Embarked']].groupby(by='Embarked')

In [48]:
group_by_embarked.count()

Sex  Fare  Age  Survived
Embarked                          
C         168   168  130       168
Q          77    77   28        77
S         644   644  554       644

In [50]:
group_by_embarked.max()

Sex      Fare   Age  Survived
Embarked                                
C         male  512.3292  71.0         1
Q         male   90.0000  70.5         1
S         male  263.0000  80.0         1

## Correlation

In [51]:
titanic_df[['Age','Fare']].corr()

Age      Fare
Age   1.000000  0.096067
Fare  0.096067  1.000000

The correlation between age and fare i 0.096067. The value of correlation lies between 0 and 1.  
Usually variables with correlation in range -0.3 to 0.3 are considered not related.  
This means that age and fare are less correlated i.e. value of age or fare does not influence the other one that much.  
Negative correlation means with increase in one variable, the other variable's value decreases. The relation is considered to be inverse.

**Sample**
-

In [59]:
titanic_df.sample(n=5,random_state = 99)

PassengerId  Survived  Pclass                                    Name  \
247          248         1       2         Hamalainen, Mrs. William (Anna)   
823          824         1       3                      Moor, Mrs. (Beila)   
566          567         0       3                    Stoytcheff, Mr. Ilia   
666          667         0       2             Butler, Mr. Reginald Fenton   
199          200         0       2  Yrois, Miss. Henriette ("Mrs Harbeck")   

        Sex   Age  SibSp  Parch  Ticket     Fare Cabin Embarked  
247  female  24.0      0      2  250649  14.5000   NaN        S  
823  female  27.0      0      1  392096  12.4750  E121        S  
566    male  19.0      0      0  349205   7.8958   NaN        S  
666    male  25.0      0      0  234686  13.0000   NaN        S  
199  female  24.0      0      0  248747  13.0000   NaN        S